### PR TITLE
Cc 63 frontend

### PR DESCRIPTION
**Pull request by falcaodiogo**
image
image

### CODE REVIEW

**Review by RGarrido03 — CHANGES_REQUESTED** [review]
Corrige isto bitch

File: package.json [hunk 1]
```diff
@@ -20,8 +20,10 @@
     "framer-motion": "^11.1.7",
     "next-themes": "^0.3.0",
     "react": "^18.2.0",
+    "react-cookie": "^7.1.4",
     "react-dom": "^18.2.0",
     "react-material-symbols": "^4.4.0",
+    "react-router-dom": "^6.23.0",
```
[the remark this answers is not in the document]
Comment (falcaodiogo) [reply package.json:26]: Done.

File: src/components/tripcard.tsx [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+            </div>
+          </div>
+        </CardBody>
+        <Divider />
+        <CardFooter>
+          {trip.freeSeats === 0 ? "No" : trip.freeSeats} seats available
+        </CardFooter>
+      </Card>
+    </Link>
+  ) : (
```
[the remark this answers is not in the document]
Comment (falcaodiogo) [reply src/components/tripcard.tsx:98]: Oh, ok!

File: src/routes/index.lazy.tsx [hunk 3]
```diff
@@ -1,13 +1,135 @@
...
+  const [searchEnabled, setSearchEnabled] = useState<boolean>(false);
+
+  const { data: cities } = useQuery<City[]>({
+    queryKey: ["cities"],
+    queryFn: () => getCities(),
+  });
+
+  useEffect(() => {
+    if (departure !== 0 && arrival !== 0) {
+      setSearchEnabled(true);
```
[the remark this answers is not in the document]
Comment (falcaodiogo) [reply src/routes/index.lazy.tsx:35]: Verdade, peço desculpa 
image

File: src/types/city.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+export interface City {
+  id: number;
+  name: string;
```
[the remark this answers is not in the document]
Comment (falcaodiogo) [reply src/types/city.ts:3]: Yup yup, changed.

File: src/routes/index.lazy.tsx [hunk 5]
```diff
@@ -1,13 +1,135 @@
...
+              }}
+            >
+              <Button
+                color="primary"
+                size="lg"
+                className="h-14"
+                isDisabled={!searchEnabled}
+                onClick={
+                  searchEnabled
+                    ? () => {
```
[the remark this answers is not in the document]
Comment (falcaodiogo) [reply src/routes/index.lazy.tsx:115]: image

**Review by RGarrido03 — APPROVED** [review]
Perfect now. Thanks!!

**Review by RGarrido03 — APPROVED** [review]
Reviewed and approved. Next!

**Pull request merged**